azavea/raster-foundry
# Add airflow tasks/rf functions to import sentinel 2 data

This is the start of adding Sentinel 2 metadata indexing as an Airflow task. At a high level, the idea is that this job would run as a regularly scheduled Airflow task that would handle importing metadata for a given period, creating thumbnails, and extracting footprints.

Luckily all of this information is available on S3 and is fairly comprehensive.

Given a year, month, and day this task will search for new tiles
uploaded for that given day. For each tile path found, an import job is
run that does the following:
 - creates a thumbnail object using the provided `preview.png` from
   sentinel 2
 - extracts the footprint and reprojects to 3857 based on information in
   `tileInfo.json`
 - create image objects based with appropriate metadata/band information
 - creates scene objects based on `tileInfo.json`

There are still a few tasks left to do with this:
 - [x] verify that interaction with the API works
 - [x] add the abiilty to poast thumbnails, images, and footprint during
 scene creation to streamline interaction
 - [x] verify backfilling works correctly and make job dynamic

## Testing Instructions

- pull branch, run migrations
- at the root of the project inside the VM run `./scripts/load_development_data`
- start the api server
- go to `/api/scenes/` and you should have 48,096 scenes
- review DAG/rf code (didn't write tests because it's likely to be in flux quite a bit for the short term)
